Checkstyle for master (5.0-SNAPSHOT) fails on Solaris

Checkstyle fail for `master` branch on Solaris. It's running by following command with Oracle JDK 8 and Maven 3.6.3:
```
clean install -U -B -DfailIfNoTests=false -DskipTests -Dmaven.test.failure.ignore=true -DvmHeapSettings="-Xms6g -Xmx6g -XX:MaxPermSize=2048m" -DextraVmArgs="-verbose:gc -Xloggc:$WORKSPACE/test_gc.log -XX:+PrintGCTimeStamps -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+PrintGCApplicationStoppedTime -XX:+PrintGCApplicationConcurrentTime"
```

Output:
```
15:08:02 [INFO] Reactor Summary for Hazelcast Root 5.0-SNAPSHOT:
15:08:02 [INFO] 
15:08:02 [INFO] Hazelcast Root ..................................... SUCCESS [  4.286 s]
15:08:02 [INFO] hazelcast .......................................... SUCCESS [05:48 min]
15:08:02 [INFO] hazelcast-spring ................................... SUCCESS [  8.525 s]
15:08:02 [INFO] hazelcast-spring-tests ............................. SUCCESS [  2.378 s]
15:08:02 [INFO] hazelcast-build-utils .............................. SUCCESS [  6.097 s]
15:08:02 [INFO] hazelcast-jet-extensions ........................... SUCCESS [  0.414 s]
15:08:02 [INFO] hazelcast-jet-kafka ................................ FAILURE [  0.244 s]
15:08:02 [INFO] hazelcast-jet-avro ................................. SKIPPED
15:08:02 [INFO] hazelcast-jet-csv .................................. SKIPPED
15:08:02 [INFO] hazelcast-jet-hadoop-core .......................... SKIPPED
15:08:02 [INFO] hazelcast-sql ...................................... SKIPPED
15:08:02 [INFO] hazelcast-all ...................................... SKIPPED
15:08:02 [INFO] hazelcast-jet-cdc-debezium ......................... SKIPPED
15:08:02 [INFO] hazelcast-jet-cdc-mysql ............................ SKIPPED
15:08:02 [INFO] hazelcast-jet-cdc-postgres ......................... SKIPPED
15:08:02 [INFO] hazelcast-jet-elasticsearch-5 ...................... SKIPPED
15:08:02 [INFO] hazelcast-jet-elasticsearch-6 ...................... SKIPPED
15:08:02 [INFO] hazelcast-jet-elasticsearch-7 ...................... SKIPPED
15:08:02 [INFO] hazelcast-jet-grpc ................................. SKIPPED
15:08:02 [INFO] hazelcast-jet-hadoop-dist .......................... SKIPPED
15:08:02 [INFO] hazelcast-jet-files-azure .......................... SKIPPED
15:08:02 [INFO] hazelcast-jet-files-gcs ............................ SKIPPED
15:08:02 [INFO] hazelcast-jet-files-s3 ............................. SKIPPED
15:08:02 [INFO] hazelcast-jet-hadoop ............................... SKIPPED
15:08:02 [INFO] hazelcast-jet-hadoop-all ........................... SKIPPED
15:08:02 [INFO] hazelcast-jet-kinesis .............................. SKIPPED
15:08:02 [INFO] hazelcast-jet-protobuf ............................. SKIPPED
15:08:02 [INFO] hazelcast-jet-python ............................... SKIPPED
15:08:02 [INFO] hazelcast-jet-s3 ................................... SKIPPED
15:08:02 [INFO] hazelcast-distribution ............................. SKIPPED
15:08:02 [INFO] ------------------------------------------------------------------------
15:08:02 [INFO] BUILD FAILURE
15:08:02 [INFO] ------------------------------------------------------------------------
15:08:02 [INFO] Total time:  06:14 min
15:08:02 [INFO] Finished at: 2021-05-24T18:17:10+03:00
15:08:02 [INFO] ------------------------------------------------------------------------
15:08:02 Waiting for Jenkins to finish collecting data
15:08:02 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.1:checkstyle (default) on project hazelcast-jet-kafka: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: Unable to find suppressions file at location: ${maven.multiModuleProjectDirectory}/checkstyle/suppressions_jet.xml: Could not find resource '${maven.multiModuleProjectDirectory}/checkstyle/suppressions_jet.xml'. -> [Help 1]
15:08:02 [ERROR] 
15:08:02 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
15:08:02 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
15:08:02 [ERROR] 
15:08:02 [ERROR] For more information about the errors and possible solutions, please read the following articles:
15:08:02 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
15:08:02 [ERROR] 
15:08:02 [ERROR] After correcting the problems, you can resume the build with the command
15:08:02 [ERROR]   mvn <goals> -rf :hazelcast-jet-kafka
```

Regarding the consecutive error, 

    [ERROR] Failed to execute goal org.xolstice.maven.plugins:protobuf-maven-plugin:0.6.1:test-compile (java) on project    
    hazelcast-jet-grpc: Unable to resolve artifact: Missing:
    [ERROR] ----------
    [ERROR] 1) com.google.protobuf:protoc:exe:sunos-sparc_64:3.13.0
    [ERROR] 
    [ERROR] Try downloading the file manually from the project website.

 Unfortunately there is no protobuf support for solaris https://github.com/protocolbuffers/protobuf/releases/tag/v3.17.3

I tried `mvn clean install` by excluding the above module

`mvn clean install -X -U -B -pl !'com.hazelcast.jet:hazelcast-jet-protobuf' -DfailIfNoTests=false -Dmaven.multiModuleProjectDirectory=${workspace} -DskipTests -Dmaven.test.failure.ignore=true -DvmHeapSettings="-Xms6g -Xmx6g -XX:MaxPermSize=2048m" -DextraVmArgs="-verbose:gc -Xloggc:$WORKSPACE/test_gc.log -XX:+PrintGCTimeStamps -XX:+PrintGCDetails -XX:+PrintGCDateStamps -XX:+PrintTenuringDistribution -XX:+PrintGCApplicationStoppedTime -XX:+PrintGCApplicationConcurrentTime""`

Unfortunately (x2 ☹️), Jenkins maven plugin does not support reactor module exclusion (i.e., not executing the goal on some modules) https://issues.jenkins.io/browse/JENKINS-26472

--- a/distribution/pom.xml
+++ b/distribution/pom.xml
@@ -31,7 +31,7 @@
 
     <properties>
         <configs.from.jar>${project.build.directory}/config</configs.from.jar>
-        <main.basedir>${maven.multiModuleProjectDirectory}</main.basedir>
+        <main.basedir>${project.parent.basedir}</main.basedir>
     </properties>
 
     <build>

--- a/extensions/cdc-debezium/pom.xml
+++ b/extensions/cdc-debezium/pom.xml
@@ -37,7 +37,7 @@
         <log4j2.slf4j.binding.version>2.13.2</log4j2.slf4j.binding.version>
 
         <!-- needed for CheckStyle -->
-        <checkstyle.headerLocation>${maven.multiModuleProjectDirectory}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
+        <checkstyle.headerLocation>${main.basedir}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
     </properties>
 
     <licenses>

--- a/extensions/cdc-mysql/pom.xml
+++ b/extensions/cdc-mysql/pom.xml
@@ -37,7 +37,7 @@
         <log4j2.slf4j.binding.version>2.13.2</log4j2.slf4j.binding.version>
 
         <!-- needed for CheckStyle -->
-        <checkstyle.headerLocation>${maven.multiModuleProjectDirectory}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
+        <checkstyle.headerLocation>${main.basedir}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
     </properties>
 
     <licenses>

--- a/extensions/cdc-postgres/pom.xml
+++ b/extensions/cdc-postgres/pom.xml
@@ -37,7 +37,7 @@
         <log4j2.slf4j.binding.version>2.13.2</log4j2.slf4j.binding.version>
 
         <!-- needed for CheckStyle -->
-        <checkstyle.headerLocation>${maven.multiModuleProjectDirectory}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
+        <checkstyle.headerLocation>${main.basedir}/checkstyle/ClassHeaderApache.txt</checkstyle.headerLocation>
     </properties>
 
     <licenses>

--- a/extensions/elasticsearch/elasticsearch-5/pom.xml
+++ b/extensions/elasticsearch/elasticsearch-5/pom.xml
@@ -34,6 +34,10 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
+    <properties>
+        <main.basedir>${project.basedir}/../../..</main.basedir>
+    </properties>
+
     <build>
         <plugins>
             <plugin>

--- a/extensions/elasticsearch/elasticsearch-6/pom.xml
+++ b/extensions/elasticsearch/elasticsearch-6/pom.xml
@@ -34,6 +34,10 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
+    <properties>
+        <main.basedir>${project.basedir}/../../..</main.basedir>
+    </properties>
+
     <build>
         <plugins>
             <plugin>

--- a/extensions/elasticsearch/elasticsearch-7/pom.xml
+++ b/extensions/elasticsearch/elasticsearch-7/pom.xml
@@ -34,6 +34,10 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
+    <properties>
+        <main.basedir>${project.basedir}/../../..</main.basedir>
+    </properties>
+
     <build>
         <plugins>
             <plugin>

--- a/extensions/hadoop-dist/pom.xml
+++ b/extensions/hadoop-dist/pom.xml
@@ -32,6 +32,10 @@
         <version>5.0-SNAPSHOT</version>
     </parent>
 
+    <properties>
+        <main.basedir>${project.basedir}/../../..</main.basedir>
+    </properties>
+
     <modules>
         <module>files-azure</module>
         <module>files-gcs</module>

--- a/extensions/hazelcast-3-connector/pom.xml
+++ b/extensions/hazelcast-3-connector/pom.xml
@@ -18,10 +18,10 @@
 
     <properties>
         <!-- needed for CheckStyle (we need to repeat this here because this module has hazelcast-root as parent) -->
-        <main.basedir>${maven.multiModuleProjectDirectory}</main.basedir>
-        <checkstyle.headerLocation>${maven.multiModuleProjectDirectory}/checkstyle/ClassHeaderHazelcastCommunity.txt</checkstyle.headerLocation>
-        <checkstyle.configLocation>${maven.multiModuleProjectDirectory}/checkstyle/checkstyle_jet.xml</checkstyle.configLocation>
-        <checkstyle.supressionsLocation>${maven.multiModuleProjectDirectory}/checkstyle/suppressions_jet.xml</checkstyle.supressionsLocation>
+        <main.basedir>${project.basedir}/../../..</main.basedir>
+        <checkstyle.headerLocation>${main.basedir}/checkstyle/ClassHeaderHazelcastCommunity.txt</checkstyle.headerLocation>
+        <checkstyle.configLocation>${main.basedir}/checkstyle/checkstyle_jet.xml</checkstyle.configLocation>
+        <checkstyle.supressionsLocation>${main.basedir}/checkstyle/suppressions_jet.xml</checkstyle.supressionsLocation>
     </properties>
 
     <dependencies>

--- a/extensions/pom.xml
+++ b/extensions/pom.xml
@@ -68,10 +68,10 @@
         </argLine>
 
         <!-- needed for CheckStyle -->
-        <main.basedir>${maven.multiModuleProjectDirectory}</main.basedir>
-        <checkstyle.headerLocation>${maven.multiModuleProjectDirectory}/checkstyle/ClassHeaderHazelcastCommunity.txt</checkstyle.headerLocation>
-        <checkstyle.configLocation>${maven.multiModuleProjectDirectory}/checkstyle/checkstyle_jet.xml</checkstyle.configLocation>
-        <checkstyle.supressionsLocation>${maven.multiModuleProjectDirectory}/checkstyle/suppressions_jet.xml</checkstyle.supressionsLocation>
+        <main.basedir>${project.basedir}/../..</main.basedir>
+        <checkstyle.headerLocation>${main.basedir}/checkstyle/ClassHeaderHazelcastCommunity.txt</checkstyle.headerLocation>
+        <checkstyle.configLocation>${main.basedir}/checkstyle/checkstyle_jet.xml</checkstyle.configLocation>
+        <checkstyle.supressionsLocation>${main.basedir}/checkstyle/suppressions_jet.xml</checkstyle.supressionsLocation>
     </properties>
 
     <licenses>

